integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/iceberg/SparkIcebergCatalogRestHiveBackendIT.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.integration.test.spark.iceberg;
+
+import com.datastrato.gravitino.auxiliary.AuxiliaryServiceManager;
+import com.datastrato.gravitino.spark.connector.iceberg.IcebergPropertiesConstants;
+import com.google.common.collect.Maps;
+import java.util.Map;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestInstance;
+
+/** This class use Iceberg RESTCatalog for test, and the real backend catalog is HiveCatalog. */
+@Tag("gravitino-docker-it")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SparkIcebergCatalogRestHiveBackendIT extends SparkIcebergCatalogIT {
+
+  private static final String icebergRestServiceName = "iceberg-rest";
+
+  @Override
+  protected Map<String, String> getCatalogConfigs() {
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    catalogProperties.put(
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_BACKEND,
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_BACKEND_REST);
+    catalogProperties.put(
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_URI, icebergRestServiceUri);
+    catalogProperties.put(
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_WAREHOUSE, warehouse);
+
+    catalogProperties.put(
+        AuxiliaryServiceManager.GRAVITINO_AUX_SERVICE_PREFIX
```

Comment:
By default, Gravitino server start with Iceberg REST service with memory backend.

Reply:
yes, i change it to Hive Backend by configurations.

Reply:
why not default memory catalog for test?

Reply:
> why not using default memory catalog for the test?

@FANNG1 Don't we require testing of all backend?

Reply:
I think testing memory catalog backend is ok,  the test test REST catalog mainly

Reply:
> I think testing memory catalog backend is ok, the test test REST catalog mainly

ok.

Reply:
now we have many test cases check the hdfs location, if we only test memory backend, these all cases will failed. cc @FANNG1

Reply:
```
org.opentest4j.AssertionFailedError: HDFS directory not exists,/tmp/default_db/new_insert_select_partition_table/data/name=gravitino_it_test/age=2 ==> expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:210)
	at com.datastrato.gravitino.integration.test.spark.SparkCommonIT.checkDirExists(SparkCommonIT.java:684)
	at com.datastrato.gravitino.integration.test.spark.SparkCommonIT.checkPartitionDirExists(SparkCommonIT.java:675)
	at com.datastrato.gravitino.integration.test.spark.SparkCommonIT.testInsertDatasourceFormatPartitionTableAsSelect(SparkCommonIT.java:663)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```

Reply:
imo, we are better to support rest with hive backend and rest with jdbc backend

Reply:
I think test one catalog backend is enough, and if not using memory, we should figure out a general way to start up Iceberg REST service.

Reply:
> I think test one catalog backend is enough, and if not using memory, we should figure out a general way to start up Iceberg REST service.

got it.

Reply:
@FANNG1 if we only test one catalog backend, such as Hive backend, how about hardcoding the configuration of IcebergRestService in `SparkEnvIT` to start IcebergRestService?

Reply:
we can just move these service configurations to SparkEnvIT because we just test hive catalog backend.

Reply:
which configurations are hardcoding?

Reply:
> which configurations are hardcoding?

https://github.com/datastrato/gravitino/blob/358c3cada4daf536f21932aaf07a3ff36b8cb8cd/integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/iceberg/SparkIcebergCatalogRestHiveBackendIT.java#L32-L49

@FANNG1 these configs.

Reply:
these configurations seems necessary to start with the hive catalog backend, I think it's ok

Reply:
> these configurations seems necessary to start with the hive catalog backend, I think it's ok

ok.

Reply:
have fixed this comment, and please help review again if you are free cc @FANNG1